antidotum/tergo/R/styling.R

```diff
@@ -21,83 +21,111 @@ style <- function(config_file = "tergo.toml", configuration = list(), ...) {
   }
   if (!is.list(configuration)) {
-    stop("Configuration must be a list")
+    stop("Configuration must be a list.")
+  }
+  if (!is.logical(verbose) || length(verbose) != 1) {
+    stop("verbose must be a single logical value.")
   }
 
-  # Read a configuration file
+  # Read Configuration File
   wd <- path
   config <- NULL
 
   repeat {
-    if (file.exists(file.path(wd, config_file))) {
-      config <- file.path(wd, config_file)
+    config_path <- file.path(wd, config_file)
+    if (file.exists(config_path)) {
+      config <- config_path
       break
     }
-    # OS agnostic file path splitting
-    if (strsplit(wd, split = .Platform$file.sep) |> length() == 1) {
+    # Stop if at the root directory
+    if (dirname(wd) == wd) {
       break
     }
     wd <- dirname(wd)
   }
+
   if (!is.null(config)) {
     config <- get_config(config)
   } else {
     config <- get_default_config()
   }
   config[names(configuration)] <- configuration
 
-  # Find package root. If not found, then just list files in the current directory
+  # Find Package Root
   package_root <- getwd()
   while (!file.exists(file.path(package_root, "DESCRIPTION"))) {
-    package_root <- dirname(package_root)
-    if (strsplit(package_root, split = .Platform$file.sep) |> length() == 1) {
+    parent_dir <- dirname(package_root)
+    if (parent_dir == package_root) { # Reached root without finding DESCRIPTION
       package_root <- getwd()
       break
     }
+    package_root <- parent_dir
   }
 
   if (!file.exists(file.path(package_root, "DESCRIPTION")) && !force) {
     message("No package detected. Exiting without formatting anything.")
     return(invisible())
   }
 
+  # List Relevant Files
   files <- list.files(package_root, recursive = TRUE, full.names = TRUE)
   files <- Filter(function(file) any(endsWith(file, extensions)), files)
-  # Format
+
+  # Define ANSI Color Codes and Unicode Symbols
+  green_tick <- "\u001B[32m\u2714\u001B[0m"
+  red_cross <- "\u001B[31m\u274C\u001B[0m"
+
+  success_count <- 0
+
   for (file in files) {
     tryCatch(
-      expr = {
-        style_file(file, configuration)
+      {
+        style_file(file, config)
+        success_count <- success_count + 1
+        # Print File Path and Green Tick
+        if (verbose) cat(sprintf("%s %s\n", basename(file), green_tick))
       },
       error = function(err) {
-        cat(sprintf("Error formatting the file: %s", file))
-        print(err)
+        # Print File Path, Red Cross, and Error Message
+        if (verbose) cat(sprintf("%s %s : %s\n", basename(file), red_cross, err$message))
       }
     )
   }
+
+  if (verbose) {
+    summary_bullet <- "\u25B6" # Black Right-Pointing Triangle
+    cat("\nSummary:\n")
+    cat(sprintf("  %s Files processed : %d\n", summary_bullet, length(files)))
+    cat(sprintf("  %s Successful     : %d\n", green_tick, success_count))
+    cat(sprintf("  %s Failed         : %d\n", red_cross, length(files) - success_count))
```

Comment:
```suggestion
    cat(sprintf("  %s Files processed : %d\n", summary_bullet, length(files)))
    cat(sprintf("  %s Successful      : %d\n", green_tick, success_count))
    cat(sprintf("  %s Failed          : %d\n", red_cross, length(files) - success_count))
```
Since we are already trying to be nice xd